packages/vitest/src/runtime/console.ts

```diff
@@ -84,7 +84,7 @@ export function createCustomConsole(defaultState?: WorkerGlobalState) {
       sendLog(type, taskId, content, buffer.length)
     }
     const timer = timers.get(taskId)!
-    buffers.set(taskId, [])
+    buffers.delete(taskId)
```

Comment:
Awesome, thanks for this fix! 🙌 
I've ran into this before but just couldn't figure out how to debug it, as any `console.log`'s would change the behaviour of logging. And stopping with debugger would change how the reporter's render loop prints. 

How did you narrow down the root cause to this line?

Reply:
Actually I didn't do debug, but just figuring out and hoping fix the repro :smile: 

What appears to be happening is the scheduling always flushes `sendStderr/sendStdout` together,

https://github.com/vitest-dev/vitest/blob/2a50464d58e98f58fed513971a570a952081bfef/packages/vitest/src/runtime/console.ts#L52-L61

For example,

```ts
console.log(1);
console.error(2);
```

becoming

```ts
sendStdout(taskId) 
sendStderr(taskId) 
```

but

```ts
console.log(1);
await new Promise(r => setTimeout(r, 10));
console.error(2);
```

becomes

```ts
// 1st flush
sendStdout(taskId) 
sendStderr(taskId)  // this is no log since `stderrBuffer.get(taskId)` is undefined

// 2nd flush
sendStdout(taskId)  // this ends up <empty line> since `stdoutBuffer.get(taskId) = []`
sendStderr(taskId) 
```

We could also just check `buffer.length > 0` above and probably other ways to refactor things around here.